chromium_src/chrome/browser/updater/scheduler_mac.cc

```diff
@@ -46,34 +47,36 @@ void CheckProcessExit(base::Process process, base::OnceClosure callback) {
 }  // namespace
 
 void DoPeriodicTasks(base::OnceClosure callback) {
-  if (!keystone_glue::KeystoneEnabled()) {
-    return;
-  }
-
-  // The registration framework doesn't provide a mechanism to ask Keystone to
-  // just do its normal routine tasks, so instead launch the agent directly.
-  // The agent can be in one of four places depending on the age and mode of
-  // Keystone.
-  for (UpdaterScope scope : {UpdaterScope::kSystem, UpdaterScope::kUser}) {
-    std::optional<base::FilePath> keystone_path = GetKeystoneFolderPath(scope);
-    if (!keystone_path) {
-      continue;
-    }
-    for (const std::string& folder : {"Helpers", "Resources"}) {
-      base::FilePath agent_path =
-          keystone_path->Append("Contents")
-              .Append(folder)
-              .Append(
-                  "GoogleSoftwareUpdateAgent.app/Contents/MacOS/"
-                  "GoogleSoftwareUpdateAgent");
-      if (base::PathExists(agent_path)) {
-        CheckProcessExit(base::LaunchProcess(base::CommandLine(agent_path), {}),
-                         std::move(callback));
-        return;
+  if (brave::ShouldUseOmaha4()) {
+    DoPeriodicTasks_ChromiumImpl(std::move(callback));
+  } else if (keystone_glue::KeystoneEnabled()) {
+    // The registration framework doesn't provide a mechanism to ask Keystone to
+    // just do its normal routine tasks, so instead launch the agent directly.
+    // The agent can be in one of four places depending on the age and mode of
+    // Keystone.
+    for (UpdaterScope scope : {UpdaterScope::kSystem, UpdaterScope::kUser}) {
+      std::optional<base::FilePath> keystone_path =
+          GetKeystoneFolderPath(scope);
+      if (!keystone_path) {
+        continue;
+      }
+      for (const std::string& folder : {"Helpers", "Resources"}) {
+        base::FilePath agent_path =
+            keystone_path->Append("Contents")
+                .Append(folder)
+                .Append(
+                    "GoogleSoftwareUpdateAgent.app/Contents/MacOS/"
+                    "GoogleSoftwareUpdateAgent");
+        if (base::PathExists(agent_path)) {
```

Comment:
`base::PathExists` is a blocking call but `DoPeriodicTasks` is not posted with `MayBlock`, which results in https://github.com/brave/brave-browser/issues/41360. I know it's not new with this PR, but I guess these conflicts are now more visible in cr130.